pkg/testutils/bazel.go

```diff
@@ -80,7 +80,14 @@ func TestSrcDir() string {
 func TestTempDir(t testing.TB) (string, func()) {
 	if RunningUnderBazel() {
 		// Bazel sets up private temp directories for each test.
-		return requireEnv(testTmpDirEnv), func() {}
+		// Normally, this private temp directory will be cleaned up automatically.
+		// However, we do use external tools (such as stress) which re-execute the
+		// same test multiple times.  Bazel, on the other hand, does not know about
+		// this, and only creates this temporary directory once.  So, ensure we create
+		// a unique temporary directory underneath bazel TEST_TMPDIR.
+		if err := os.Setenv("TMPDIR", requireEnv(testTmpDirEnv)); err != nil {
```

Comment:
Staring at this a bit further, I think what we want instead is to pass in whatever `TEST_TMPDIR` points to into the empty base directory here:

https://github.com/cockroachdb/cockroach/blob/cc4300cc685fc6248fe1cd017a3cbef1922d50f1/pkg/testutils/dir.go#L24

What we're doing now by setting TMPDIR directly seems a round about way of achieving the same thing. What do you think?

Reply:
Done (I drop TestTempDir altogether, but had to update callers).